### DISCUSSION TITLE
Change Font globally

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#14018] [Q&A] [answered]
Is there a way to change the font globally in a way that is includes tooltips, flyouts and similar controls?

Background: It tried setting the font on my main Window and many subcontrols inherited the setting but not all.

**Comment by thevortexcloud** [comment]
Did you try setting an application style in your `App.axaml`?

**Comment by DevAM-Tools** [comment]
What exactly do you mean? Application does not have a bindable property for the font.

Setting the font for each individual control type is a kind of workaround. However, this has the disadvantage that the list always has to be kept up to date, depending on which controls you use.

**Comment by ahopper** [comment]
something like
```<FontFamily x:Key="ContentControlThemeFontFamily">avares://MyApp/Assets#MyFont</FontFamily>```
in app.axml resources should do it if you are using the fluent theme.

**Comment by DevAM-Tools** [comment, marked as answer]
That seems to do the job: (without a massive amount of testing yet)

```xaml
<Application xmlns="https://github.com/avaloniaui"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    RequestedThemeVariant="Dark">

    <Application.Resources>
        <FontFamily x:Key="Font">avares://[... Path to the font file]</FontFamily>
    </Application.Resources>

    <Application.Styles>
        <FluentTheme />
	<Style Selector="ContentPresenter">
		<Setter Property="FontFamily" Value="{StaticResource Font}" />
	</Style>
    </Application.Styles>
</Application>
```
